src/crud_ops.rs

```diff
@@ -143,6 +225,49 @@ impl Client {
         counter!(metrics::total_delete_ops).increment(1);
         with_retries!(self, self.delete_impl(path).await)
     }
+
+    async fn bulk_delete_impl(&self, paths: &Vec<Path>) -> crate::Result<Vec<(Path, String)>> {
+        let stream = stream::iter(paths.iter().map(|path| Ok(path.clone()))).boxed();
+        // Counter to keep track of the index of the path that failed to delete
+        let counter = Arc::new(AtomicUsize::new(0));
+        let bulk_failed_entries = self.store.delete_stream(stream)
+            .filter_map(|result| async {
+                let counter_clone = Arc::clone(&counter);
+                let index = counter_clone.fetch_add(1, Ordering::SeqCst);
```

Comment:
The `StreamExt` trait has the `enumerate` method that already gives us the index we want below, so we can get rid of the shared counter:
```suggestion
        let bulk_failed_entries = self.store.delete_stream(stream)
            .enumerate()
            .filter_map(|(index, result)| async {
```

Reply:
I am using the shared counter to count the number of callbacks and decide whether we received a generic error